Lib/test/test_type_annotations.py

```diff
@@ -457,3 +457,156 @@ class format: pass
             "cannot access free variable 'format' where it is not associated with a value in enclosing scope",
         ):
             ns["f"].__annotations__
+
+
+class ConditionalAnnotationTests(unittest.TestCase):
+    def check_scopes(self, code, true_annos, false_annos):
+        for scope in ("class", "module"):
+            for (cond, expected) in ((True, true_annos), (False, false_annos)):
+                with self.subTest(scope=scope, cond=cond):
+                    code_to_run = code.format(cond=cond)
+                    if scope == "class":
+                        code_to_run = "class Cls:\n" + textwrap.indent(textwrap.dedent(code_to_run), " " * 4)
+                    ns = run_code(code_to_run)
+                    if scope == "class":
+                        self.assertEqual(ns["Cls"].__annotations__, expected)
+                    else:
+                        self.assertEqual(ns["__annotate__"](annotationlib.Format.VALUE),
+                                         expected)
+
```

Comment:
Do we also want a test for `match..case`?

Reply:
Good idea, I forgot about match